crates/autopilot/src/domain/auction/order.rs

```diff
@@ -43,6 +39,25 @@ impl Order {
     }
 }
 
+// uid as 56 bytes: 32 for orderDigest, 20 for ownerAddress and 4 for validTo
+#[derive(Copy, Clone, PartialEq, Hash, Eq)]
+pub struct OrderUid(pub [u8; 56]);
+
+impl OrderUid {
+    pub fn owner(&self) -> eth::Address {
+        self.parts().1.into()
+    }
+
+    /// Splits an order UID into its parts.
+    fn parts(&self) -> (H256, H160, u32) {
```

Comment:
This is dangerous because we have potentially panic accesses (using `[]`) for an unchecked inter type (`pub [u8; 56]`).  We could drop the `pub` and make a `pub fn new_checked() -> Result<T, E>`, doing a length check before even assembling `OrderUid`.

Reply:
This is a copy pasted function from model::OrderUid which served us well so far so probably not worth introducing an error handling that will never be hit. 

This function is supposed to never change/update.